src/utils/znsUtils.ts

```diff
@@ -298,3 +304,20 @@ export function fromBech32Address(
 
   return toChecksumAddress(buf.toString('hex'));
 }
+/**
+ * Parses object in format { "key.key2.key3" : value } into { key: { key2: {key3: value } } }
```

Comment:
This helper is called only in `Zns.structureResolverRecords` function. Which is called only in `Zns.resolve`.. and `Zns.resolve` is used only in the `Zns.owner` source code. It seems like there's a lot of redundant things going on.